src/python/pants/backend/python/lint/pylint/rules_integration_test.py

```diff
@@ -0,0 +1,160 @@
+    self.create_file(
+      relpath=file_content.path, contents=file_content.content.decode()
+    )
+
+  @classmethod
+  def alias_groups(cls) -> BuildFileAliases:
+    return BuildFileAliases(targets={'python_library': PythonLibrary})
+
+  @classmethod
+  def rules(cls):
+    return (*super().rules(), *pylint_rules(), RootRule(PylintTarget))
+
+  def run_pylint(
+    self,
+    source_files: List[FileContent],
+    *,
+    config: Optional[str] = None,
+    passthrough_args: Optional[str] = None,
+    interpreter_constraints: Optional[str] = None,
+    skip: bool = False,
+    origin: Optional[OriginSpec] = None,
+  ) -> LintResult:
+    args = ["--backend-packages2=pants.backend.python.lint.pylint"]
+    if config:
+      # TODO: figure out how to get this file to exist...
+      self.create_file(relpath="pylintrc", contents=config)
+      args.append("--pylint-config=pylintrc")
+    if passthrough_args:
+      args.append(f"--pylint-args='{passthrough_args}'")
+    if skip:
+      args.append(f"--pylint-skip")
+    input_snapshot = self.request_single_product(Snapshot, InputFilesContent(source_files))
+    adaptor = PythonTargetAdaptor(
+      sources=EagerFilesetWithSpec(self.source_root, {'globs': []}, snapshot=input_snapshot),
+      address=Address.parse(f"{self.source_root}:target"),
+      compatibility=[interpreter_constraints] if interpreter_constraints else None,
+    )
+    if origin is None:
+      origin = SingleAddress(directory="test", name="target")
+    target = PylintTarget(PythonTargetAdaptorWithOrigin(adaptor, origin))
+    return self.request_single_product(
+      LintResult, Params(target, create_options_bootstrapper(args=args)),
+    )
+
+  def test_single_passing_source(self) -> None:
+    self.create_python_library()
+    self.write_file(self.good_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_single_failing_source(self) -> None:
+    self.create_python_library()
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source])
+    assert result.exit_code == 16
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  def test_mixed_sources(self) -> None:
+    self.create_python_library()
+    self.write_file(self.good_source)
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.good_source, self.bad_source])
+    assert result.exit_code == 16
+    assert "good.py" not in result.stdout
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  def test_precise_file_args(self) -> None:
+    self.create_python_library()
+    self.write_file(self.good_source)
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+  
+  @pytest.mark.skip(reason="Get config file creation to work with options parsing")
```

Comment:
I believe every skipped test should point to a specific open issue -- is there a ticket tracking this yet?

Reply:
No ticket. This is my fault from when I added the options bootstrapper test utility. The issue is that our parser validates that all files from the config are in the V2 test chroot, which they are not. Instead, the files are in the temporary build root set up by TestBase, which is where they belong. The solution is to tweak the parser’s file validation logic, which I have a WIP for.

Reply:
I would love for that WIP to be posted and referenced here so that this issue doesn't get lost without context. For longer comments in general ensuring followup comments or PRs before landing is a good practice -- for skipped tests, I consider it absolutely necessary.